### ISSUE TITLE
Killing a Terminal does not stop running processes.

### ISSUE THREAD

**Issue by jsj1027** [microsoft/vscode#114286]
Issue Type: <b>Bug</b>

1. Run any long running process. For example, building vscode.
2. Hit the kill terminal trash bin Icon.
3. Check task manager/any process or resource monitering application.
4. Verify that the ran process, is still running in the backgroud. 

VS Code version: Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:32:10.090Z)  & Code 1.52.1(ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:34:46:910Z)
OS version: Linux x64 5.8.0-7630-generic & Windows_NT x64 10.0.171.34

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 3587)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: unavailable_off<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 0, 0|
|Memory (System)|15.53GB (9.36GB free)|
|Process Argv|--no-sandbox --unity-launch --crash-reporter-id c3c56e58-2e8f-40b4-b7fc-4b118b51ddd2|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|pop|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|pop|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-profile-switcher|aar|0.3.3
bracket-pair-colorizer-2|Coe|0.2.0
cpptools|ms-|1.1.3
vscodeintellicode|Vis|1.2.10
vscode-icons|vsc|11.1.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
openlogontheside:30221877
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30211401
wsl2promptcf:30224613
pythonvsdeb440:30242242
unusedprompt:30224610
folderexplorercf:30224615
openfilemenu:30224647
pythonvsded773cf:30236630
pythonvspyt600:30241727
vsjup459cf:30237188

```

</details>

Does this issue occur when all extensions are disabled?: Yes

**Comment by sbatten** [comment]
@jsj1027 building vscode with the repo's default build task is intentionally created to run even if the terminal is closed to handle window reloading. 

can you verify that running another long running task will result in this behavior? e.g. `yarn watch` (not `yarn watchd`)

**Comment by jsj1027** [comment]
So upon further investigation, I was unable to recreate the bug with a different log running process such as running make in a large C++ project.

Note: I did not test this with building vscode that was just an example. That I gave because where Im seeing this bug is actually in a propriety project im working on that I can't give many details about.

What ive come to understand about this "bug" is that it is killing the process, unless the process has some sort of mechanism to continue running. Which the process that was giving me trouble does have.
